sample_files/pre-commit-13.0/.pre-commit-config.yaml

```diff
@@ -0,0 +1,62 @@
+exclude: "^setup/"
+default_language_version:
+  python: python3
+repos:
+- repo: https://github.com/ambv/black
+  rev: 19.3b0
+  hooks:
+  - id: black
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.3.0
+  hooks:
+  - id: trailing-whitespace
+    # exclude autogenerated files
+    exclude: /README\.rst$|\.pot?$
+  - id: end-of-file-fixer
+    # exclude autogenerated files
+    exclude: /README\.rst$|\.pot?$
+  - id: debug-statements
+  - id: flake8
+    name: flake8 except __init__.py
+    exclude: /__init__\.py$
+    additional_dependencies: ["flake8-bugbear==19.8.0"]
+  - id: flake8
+    name: flake8 only __init__.py
+    args: ["--extend-ignore=F401"]  # ignore unused imports in __init__.py
+    files: /__init__\.py$
+    additional_dependencies: ["flake8-bugbear==19.8.0"]
```

Comment:
AFAIK Black enforces an opinionated version of PEP8, and flake8 checks that it's respected. However... why do we need flake8 then? Imagine a scenario where for some weird reason Black formats a file in a way that flake8 dislikes. Satisfying both linters would be confusing and time-consuming. It would be much easier to just let Black do its thing and open an issue in the Black project in case we see there's something bad.

So, can't we just drop flake8?

Reply:
In some cases Black cannot enforce it (eg a long string will not be split by Black). Also, flake8 does additional useful checks than pep8.

Reply:
If black can't fix a line because it's too long and it doesn't know where to cut it (very long word, ...), it will not change it, and as it didn't change it, it reports a success status.